<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="generator" content="hevea 2.01">
<style type="text/css">
.li-itemize{margin:1ex 0ex;}
.li-enumerate{margin:1ex 0ex;}
.dd-description{margin:0ex 0ex 1ex 4ex;}
.dt-description{margin:0ex;}
.toc{list-style:none;}
.footnotetext{margin:0ex; padding:0ex;}
div.footnotetext P{margin:0px; text-indent:1em;}
.thefootnotes{text-align:left;margin:0ex;}
.dt-thefootnotes{margin:0em;}
.dd-thefootnotes{margin:0em 0em 0em 2em;}
.footnoterule{margin:1em auto 1em 0px;width:50%;}
.caption{padding-left:2ex; padding-right:2ex; margin-left:auto; margin-right:auto}
.title{margin:2ex auto;text-align:center}
.titlemain{margin:1ex 2ex 2ex 1ex;}
.titlerest{margin:0ex 2ex;}
.center{text-align:center;margin-left:auto;margin-right:auto;}
.flushleft{text-align:left;margin-left:0ex;margin-right:auto;}
.flushright{text-align:right;margin-left:auto;margin-right:0ex;}
div table{margin-left:inherit;margin-right:inherit;margin-bottom:2px;margin-top:2px}
td table{margin:auto;}
table{border-collapse:collapse;}
td{padding:0;}
.cellpadding0 tr td{padding:0;}
.cellpadding1 tr td{padding:1px;}
pre{text-align:left;margin-left:0ex;margin-right:auto;}
blockquote{margin-left:4ex;margin-right:4ex;text-align:left;}
td p{margin:0px;}
.boxed{border:1px solid black}
.textboxed{border:1px solid black}
.vbar{border:none;width:2px;background-color:black;}
.hbar{border:none;height:2px;width:100%;background-color:black;}
.hfill{border:none;height:1px;width:200%;background-color:black;}
.vdisplay{border-collapse:separate;border-spacing:2px;width:auto; empty-cells:show; border:2px solid red;}
.vdcell{white-space:nowrap;padding:0px; border:2px solid green;}
.display{border-collapse:separate;border-spacing:2px;width:auto; border:none;}
.dcell{white-space:nowrap;padding:0px; border:none;}
.dcenter{margin:0ex auto;}
.vdcenter{border:solid #FF8000 2px; margin:0ex auto;}
.minipage{text-align:left; margin-left:0em; margin-right:auto;}
.marginpar{border:solid thin black; width:20%; text-align:left;}
.marginparleft{float:left; margin-left:0ex; margin-right:1ex;}
.marginparright{float:right; margin-left:1ex; margin-right:0ex;}
.theorem{text-align:left;margin:1ex auto 1ex 0ex;}
.part{margin:2ex auto;text-align:center}
</style><link rel="stylesheet" type="text/css" href="../../beast2.css">
<title>MEPs</title>
</head>
<body ><div id="container">
<!--HEVEA command line is: hevea MEPs -->
<!--CUT STYLE article--><!--CUT DEF section 1 --><table class="title"><tr><td style="padding:1ex"><h1 class="titlemain">Measurably evolving populations <a id="chap.MEP"></a></h1><h3 class="titlerest">Alexei J Drummond, Remco Bouckaert and Walter Xie</h3></td></tr>
</table>
<!--TOC section id=sec1 Time-stamped data-->
<h2 id="sec1" class="section">1&#XA0;&#XA0;Time-stamped data</h2><!--SEC END --><p>This tutorial estimates the rate of evolution from a set of virus sequences which have been isolated at different points in time (heterochronous or time-stamped data). The data are 129 sequences from the G (attachment protein) gene of human respiratory
syncytial virus subgroup A (RSVA) from various parts of the world with isolation dates ranging from 1956-2002 [<a href="#Zlateva%3A2004uq">2</a>, <a href="#Zlateva%3A2005qy">1</a>].
RSVA causes infections of the lower respiratory tract causing symptoms that are often indistinguishable from the common cold. By age 3, nearly all children will be infected and a small percentage (&lt;3%) will develop more serious inflammation of the bronchioles requiring hospitalisation.</p><p>The aim of this tutorial is to obtain estimates for :</p><ul class="itemize"><li class="li-itemize">
the rate of molecular evolution
</li><li class="li-itemize">the date of the most recent common ancestor
</li><li class="li-itemize">the phylogenetic relationships with measures of statistical support.
</li></ul><p>The following software will be used in this tutorial:</p><ul class="itemize"><li class="li-itemize">
<span style="font-weight:bold">BEAST</span> - this package contains the BEAST program, BEAUti, DensiTree, TreeAnnotator and other utility programs. This tutorial is written for BEAST v2.2.x, which has support for multiple partitions. It is available for download from <br>
<span style="font-family:monospace">http://www.beast2.org</span>.
</li><li class="li-itemize"><span style="font-weight:bold">Tracer</span> - this program is used to explore the output of BEAST (and other Bayesian MCMC programs). It graphically and
quantitively summarizes the distributions of continuous parameters and provides diagnostic information. At the time of
writing, the current version is v1.6. It is available for download from <span style="font-family:monospace">http://beast.bio.ed.ac.uk/</span>.
</li><li class="li-itemize"><span style="font-weight:bold">FigTree</span> - this is an application for displaying and printing molecular phylogenies, in particular those obtained using
BEAST. At the time of writing, the current version is v1.4.2. It is available for download from <span style="font-family:monospace">http://tree.bio.ed.ac.uk/</span>.
</li></ul><!--TOC subsection id=sec2 The NEXUS alignment-->
<h3 id="sec2" class="subsection">The NEXUS alignment</h3><!--SEC END --><p>
The data is in a file called <span style="font-family:monospace">RSV2.nex</span> and you can find it in the <span style="font-family:monospace">examples/nexus</span> directory in the directory where BEAST was installed. This file contains an alignment of 129 sequences from the G gene of RSVA virus, 629 nucleotides in length. Import this alignment into BEAUti. Because this is a protein-coding gene we are going to split the alignment into three partitions representing each of the three codon positions. To do this we will click the <span style="font-weight:bold">Split</span> button at the bottom of the <span style="font-weight:bold">Partitions</span> panel and then select the &#X201C;1 + 2 + 3 frame 3&#X201D; from the drop-down menu. </p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_split.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>This signifies that the first full codon starts at the third nucleotide in the alignment. This will create three rows in the partitions panel. You will have to re-link the tree and clock models across the three partitions (and name them &#X201C;tree&#X201D; and &#X201C;clock&#X201D; respectively) before continuing to the next step. The partition panel should now look something like this:</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_partition.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>By default all the taxa are assumed to have a date of zero (i.e. the sequences are assumed to be sampled at the same time).
In this case, the RSVA sequences have been sampled at various dates going back to the 1950s. The actual year of sampling
is given in the name of each taxon and we could simply edit the value in the Date column of the table to reflect these.
However, if the taxa names contain the calibration information, then a convenient way to specify the dates of the sequences
in BEAUti is to click the checkbox <span style="font-weight:bold">Use tip dates</span> and then use the <span style="font-weight:bold">Guess</span> button at the top of the <span style="font-weight:bold">Tip Dates</span> panel. Clicking this will make a dialog box appear.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_GuessDates.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>Select the option to <span style="font-style:italic">use everything</span>, choose &#X201C;after last&#X201D; from from drop-down box and type&#X2018;s&#X2019; into the corresponding text box. This will extract the trailing numbers from the taxon names after the last little &#X2019;s&#X2019;, which are interpreted as the year (in this case since 1900) that the sample was isolated.</p><p>The dates panel should now look something like this:</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_dates.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><!--TOC subsection id=sec3 Setting the substitution model-->
<h3 id="sec3" class="subsection">Setting the substitution model</h3><!--SEC END --><p>
We will use the HKY model with empirical base frequencies for all three partitions. To do this first link the site partitions and then choose HKY and Empirical from the Subst Model and Frequencies drop-boxes. Also check the estimate box for the Mutation Rate,which will finally trigger to check the &#X201C;Fix mean mutation rate&#X201D; box. </p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_Site_Model.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>Then, hold &#X2018;shift&#X2019; key to select all site models on the left side, and click <span style="font-weight:bold">OK</span> to clone the setting from defined site model (Figure <a href="#fig%3AcloneFrom">1</a>). Go through each site model, as you can see, their configurations are same now. </p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/cloneFrom.hevea.png">
<div class="caption"><table style="border-spacing:6px;border-collapse:separate;" class="cellpading0"><tr><td style="vertical-align:top;text-align:left;" >Figure 1: clone configuration from one site model to others.</td></tr>
</table></div>

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote>
<!--TOC subsubsection id=sec4 Priors -->
<h4 id="sec4" class="subsubsection">1.0.1&#XA0;&#XA0;Priors </h4><!--SEC END --><p>To set up the priors, select the &#X201C;Priors&#X201D; tab.
Choose &#X201C;Coalescent Constant Population&#X201D; for the tree prior. Set the prior on the clockRate parameter to a log-normal with <span style="font-style:italic">M</span>=&#X2212;5 and <span style="font-style:italic">S</span>=1.25. </p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_priors.hevea.png">
<a id="fig:BEAUti_priors"></a>
<div class="center"><hr style="width:80%;height:2"></div></div></blockquote>
<!--TOC subsection id=sec5 Setting the MCMC options-->
<h3 id="sec5" class="subsection">1.1&#XA0;&#XA0;Setting the MCMC options</h3><!--SEC END --><p>For this dataset let&#X2019;s initially set the chain length to <U>2,000,000</U> as this will run 
reasonably quickly on most modern computers. Set the sampling frequencies for the screen to
to <U>10,000</U>, the trace log file to <U>400</U> and the trees file to <U>400</U>.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_mcmc.hevea.png">
<a id="fig:BEAUti_mcmc"></a>
<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><!--TOC subsection id=sec6 Running BEAST-->
<h3 id="sec6" class="subsection">Running BEAST</h3><!--SEC END --><p>Save the BEAST file (e.g. <span style="font-family:monospace">RSV2.xml</span>) and run it in BEAST.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAST.hevea.png">
<div class="caption"><table style="border-spacing:6px;border-collapse:separate;" class="cellpading0"><tr><td style="vertical-align:top;text-align:left;" >Figure 2: A screenshot of BEAST.</td></tr>
</table></div>

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>Now run BEAST and when it asks for an input file, provide your newly
created XML file as input. BEAST will then run until it has finished
reporting information to the screen. The actual results files are
save to the disk in the same location as your input file. The output to the screen will
look something like this: </p><pre class="verbatim"><span style="font-size:xx-small">
             BEAST v2.2.0, 2002-2014
       Bayesian Evolutionary Analysis Sampling Trees
                 Designed and developed by
Remco Bouckaert, Alexei J. Drummond, Andrew Rambaut and Marc A. Suchard
                              
               Department of Computer Science
                   University of Auckland
                  remco@cs.auckland.ac.nz
                  alexei@cs.auckland.ac.nz
                              
             Institute of Evolutionary Biology
                  University of Edinburgh
                     a.rambaut@ed.ac.uk
                              
              David Geffen School of Medicine
           University of California, Los Angeles
                     msuchard@ucla.edu
                              
                Downloads, Help &amp; Resources:
                     http://beast2.org/
                              
Source code distributed under the GNU Lesser General Public License:
              http://github.com/CompEvol/beast2
                              
                     BEAST developers:
 Alex Alekseyenko, Trevor Bedford, Erik Bloomquist, Joseph Heled, 
 Sebastian Hoehna, Denise Kuehnert, Philippe Lemey, Wai Lok Sibon Li, 
 Gerton Lunter, Sidney Markowitz, Vladimir Minin, Michael Defoin Platel, 
           Oliver Pybus, Chieh-Hsi Wu, Walter Xie
                              
                         Thanks to:
     Roald Forsberg, Beth Shapiro and Korbinian Strimmer

... ...

        1999000     -6131.9540        80.2        -5496.4810      -635.4730 2m3s/Msamples
        2000000     -6142.7389        80.7        -5506.2818      -636.4570 2m3s/Msamples

Operator                                                Tuning    #accept    #reject      Pr(m)  Pr(acc|m)
ScaleOperator(treeScaler.t:tree)                        0.7224        546      71568     0.0360     0.0076 Try setting scaleFactor to about 0.85
ScaleOperator(treeRootScaler.t:tree)                    0.6019       5076      66818     0.0360     0.0706 Try setting scaleFactor to about 0.776
Uniform(UniformOperator.t:tree)                              -     387274     334094     0.3601     0.5369 
SubtreeSlide(SubtreeSlide.t:tree)                       3.9556      57963     302626     0.1801     0.1607 
Exchange(narrow.t:tree)                                      -      88179     271422     0.1801     0.2452 
Exchange(wide.t:tree)                                        -        153      71541     0.0360     0.0021 
WilsonBalding(WilsonBalding.t:tree)                          -        365      71560     0.0360     0.0051 
ScaleOperator(StrictClockRateScaler.c:clock)            0.7869      17571      54573     0.0360     0.2436 
UpDownOperator(strictClockUpDownOperator.c:clock)       0.7594        673      70876     0.0360     0.0094 Try setting scaleFactor to about 0.871
ScaleOperator(KappaScaler.s:RSV2_1)                     0.4069        581       1743     0.0012     0.2500 
DeltaExchangeOperator(FixMeanMutationRatesOperator)     0.3833       8247      39773     0.0240     0.1717 
ScaleOperator(KappaScaler.s:RSV2_2)                     0.4276        583       1790     0.0012     0.2457 
ScaleOperator(KappaScaler.s:RSV2_3)                     0.4421        589       1774     0.0012     0.2493 
ScaleOperator(PopSizeScaler.t:tree)                     0.5907      18802      53241     0.0360     0.2610 

     Tuning: The value of the operator's tuning parameter, or '-' if the operator can't be optimized.
    #accept: The total number of times a proposal by this operator has been accepted.
    #reject: The total number of timEnd likelihood: -6142.738945357851
es a proposal by this operator has been rejected.
      Pr(m): The probability this operator is chosen in a step of the MCMC (i.e. the normalized weight).
  Pr(acc|m): The acceptance probability (#accept as a fraction of the total proposals for this operator).


Total calculation time: 249.968 seconds
</span></pre><!--TOC subsection id=sec7 Analysing the BEAST output-->
<h3 id="sec7" class="subsection">Analysing the BEAST output</h3><!--SEC END --><p>Note that the effective sample sizes (ESSs) for many of the logged quantities are small (ESSs less than 100 will be highlighted in red by Tracer).
This is not good. A low ESS means that the trace contains a lot of correlated samples and thus may not represent the
posterior distribution well. In the bottom right of the window is a frequency plot of the samples which is expected given the
low ESSs is extremely rough.</p><p>If we select the tab on the right-hand-side labelled &#X2018;Trace&#X2019; we can view the raw trace, that is, the sampled values against the step in the MCMC chain.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/Tracer1.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>Here you can see how the samples are correlated. There are 5000 samples in the trace (we ran the MCMC for <U>2,000,000</U>
steps sampling every <U>400</U>) but adjacent samples often tend to have similar values. The ESS for the absolute rate of evolution (clockRate) is about 62 so we are only getting 1 independent sample to every 81=5000/62 actual samples). With a short run such as this one, it may also be the case that the default burn-in of 10% of the chain length is inadequate. Not excluding enough of the start of the chain as burn-in will render estimates of ESS unreliable.</p><p>The simple response to this situation is that we need to run the chain for longer. Given the lowest ESS (for the constant coalescent) is 44, it
would suggest that we have to run the chain for at least 5 times the length to get reasonable ESSs that are &gt;200. 
So let&#X2019;s go for a chain length of <U>10,000,000</U> and log every <U>2,000</U>. Go back to the <span style="font-weight:bold">MCMC</span> options section in BEAUti, and create a new BEAST XML file with a longer chain length. Now run BEAST and load the new log file into Tracer (you can leave the old one loaded
for comparison). </p><p>Click on the Trace tab and look at the raw trace plot.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/Tracer2.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>Again we have chosen options that produce 5000 samples and with an ESS of about 342 there is still auto-correlation
between the samples but &gt;340 effectively independent samples will now provide a very good estimate of the posterior distribution.
There are no obvious trends in the plot which would suggest that the MCMC has not yet converged, and there are no significant long range 
fluctuations in the trace which would suggest poor mixing.</p><p>As we are satisfied with the mixing we can now move on to one of the parameters of interest:
substitution rate. Select <span style="font-family:monospace">clockRate</span> in the left-hand table. This is the average substitution rate across all sites in the
alignment. Now choose the density plot by selecting the tab labeled <span style="font-style:italic">Marginal Density</span>. This shows a plot of the marginal posterior probability
density of this parameter. You should see a plot similar to this:</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/Tracer_density.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>As you can see the posterior probability density is roughly bell-shaped. There is some sampling noise which would be
reduced if we ran the chain for longer or sampled more often but we already have a good estimate of the mean and HPD interval. You can overlay
the density plots of multiple traces in order to compare them (it is up to the user to determine whether they are comparable on the the same axis or not). Select the relative substitution rates for all three codon positions in the table to the left (labelled
<span style="font-family:monospace">mutationRate.1</span>, <span style="font-family:monospace">mutationRate.2</span> and <span style="font-family:monospace">mutationRate.3</span>). You will now see the posterior probability densities for the relative
substitution rate at all three codon positions overlaid:</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/Tracer_relativeRates.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><!--TOC subsection id=sec8 Summarizing the trees-->
<h3 id="sec8" class="subsection">Summarizing the trees</h3><!--SEC END --><p>Use the program TreeAnnotator to summarize the tree and view the results in Figtree (Figure <a href="#fig%3ARSV2tree">3</a>).</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/RSV2_mcc_tree.hevea.png">
<div class="caption"><table style="border-spacing:6px;border-collapse:separate;" class="cellpading0"><tr><td style="vertical-align:top;text-align:left;" >Figure 3: The Maximum clade credibility tree for the G gene of 129 RSVA-2 viral samples. </td></tr>
</table></div>

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/DensiTree.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>DensiTree with clade height bars for clades with over 50% support.
Root canal tree represents maximum clade credibility tree.</p><!--TOC subsection id=sec9 Questions-->
<h3 id="sec9" class="subsection">Questions</h3><!--SEC END --><p><span style="font-style:italic">In what year did the common ancestor of all RSVA viruses sampled live? What is the
95% HPD?</span></p><!--TOC subsection id=sec10 Bonus section: Bayesian Skyline plot-->
<h3 id="sec10" class="subsection">Bonus section: Bayesian Skyline plot</h3><!--SEC END --><p>We can reconstruct the population history using the Bayesian Skyline plot. In order to do so,
load the XML file into BEAUti, select the priors-tab and change the tree prior from 
coalescent with constant population size to coalescent with Bayesian skyline.
Note that an extra item is added to the priors called &#X2018;Markov chained population sizes&#X2019;
which is a prior that ensures dependence between population sizes.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_priors2.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>By default the number of groups used in the skyline analysis is set to 5, To change this,
select menu View/Show Initialization panel and a list of parameters is shown. Select
<span style="font-family:monospace">bPopSizes.t:tree</span> and change the dimension to 3. Likewise, selection <span style="font-family:monospace">bGroupSizes.t:tree</span>
and change its dimension to 3. The dimensions of the two parameters should be the same.
More groups mean more population changes can be detected, but it also means more parameters
need to be estimated and the chain runs longer. The extended Bayesian skyline plot
automatically detects the number of changes, so it could be used as an alternative
tree prior.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/BEAUti_init.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>This analysis requires a bit longer to converge, so change the MCMC chain length to 10 million,
and the log intervals for the trace-log and tree-log to 10 thousand. 
Then, save the file and run BEAST.</p><p>To plot the population history, load the log file in tracer and select the menu 
Analysis/Bayesian Skyline Reconstruction.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/tracerBSP1.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>A dialog is shown where you can specify the tree file associated with the log file.
Also, since the youngest sample is from 2002, change the entry for age of youngest
tip to 2002.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/tracerBSP2.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><p>After some calculation, a graph appears showing population history where the median
and 95% HPD intervals are plotted. After selecting the &#X2018;solid interval&#X2019; checkbox, the
graph should look something like this.</p><blockquote class="figure"><div class="center"><div class="center"><hr style="width:80%;height:2"></div>
	
<img src="figures/tracerBSP3.hevea.png">

<div class="center"><hr style="width:80%;height:2"></div></div></blockquote><!--TOC subsection id=sec11 Questions-->
<h3 id="sec11" class="subsection">Questions</h3><!--SEC END --><ol class="enumerate" type=1><li class="li-enumerate">
<span style="font-style:italic">By what amount did the effective population size of RSVA grow from 1970 to 2002 according to the BSP?</span></li><li class="li-enumerate"><span style="font-style:italic">What are the underlying assumptions of the BSP? Are the violated by this data set?</span></li></ol>
<!--TOC subsection id=sec12 Exercise-->
<h3 id="sec12" class="subsection">1.2&#XA0;&#XA0;Exercise</h3><!--SEC END --><p>
Change the Bayesian skyline prior to extended Bayesian skyline plot (EBSP) prior and run
till convergence. EBSP produces an extra log file, called EBSP.$(seed).log where $(seed)
is replaced by the seed you used to run BEAST.
A plot can be created by running the EBSPAnalyser utility, and loading the output file
in a spreadsheet.</p><p>How many groups are indicated by the EBSP analysis?
This is much lower than for BSP. How does this affect the population history plots?</p><!--TOC section id=sec13 References-->
<h2 id="sec13" class="section">References</h2><!--SEC END --><dl class="thebibliography"><dt class="dt-thebibliography">
<a id="Zlateva:2005qy">[1]</a></dt><dd class="dd-thebibliography">
Kalina&#XA0;T Zlateva, Philippe Lemey, Elien Mo&#XEB;s, Anne-Mieke Vandamme, and Marc
Van&#XA0;Ranst, <em>Genetic variability and molecular evolution of the human
respiratory syncytial virus subgroup b attachment g protein</em>, J Virol
<span style="font-weight:bold">79</span> (2005), no.&#XA0;14, 9157&#X2013;67.</dd><dt class="dt-thebibliography"><a id="Zlateva:2004uq">[2]</a></dt><dd class="dd-thebibliography">
Kalina&#XA0;T Zlateva, Philippe Lemey, Anne-Mieke Vandamme, and Marc Van&#XA0;Ranst,
<em>Molecular evolution and circulation patterns of human respiratory
syncytial virus subgroup a: positively selected sites in the attachment g
glycoprotein</em>, J Virol <span style="font-weight:bold">78</span> (2004), no.&#XA0;9, 4675&#X2013;83.</dd></dl><!--CUT END -->
<!--HTMLFOOT-->
<!--ENDHTML-->
<!--FOOTER-->
<hr style="height:2"><blockquote class="quote"><em>This document was translated from L<sup>A</sup>T<sub>E</sub>X by
</em><a href="http://hevea.inria.fr/index.html"><em>H</em><em><span style="font-size:small"><sup>E</sup></span></em><em>V</em><em><span style="font-size:small"><sup>E</sup></span></em><em>A</em></a><em>.</em></blockquote></div></body>
</html>
